src/components/Image/Detail/ImageDetailProvider.tsx

```diff
@@ -88,6 +93,35 @@ export function ImageDetailProvider({
   }
 
   const index = findCurrentImageIndex();
+
+  const updateImage = (id: number, data: Partial<ImagesInfiniteModel>) => {
+    queryUtils.image.getInfinite.setInfiniteData(
+      { ...filters, userId: !!reactions?.length ? undefined : userId, postId, browsingLevel },
+      produce((queryData) => {
+        if (!queryData?.pages?.length) return;
+
+        for (const page of queryData.pages)
+          for (const item of page.items) {
+            if (item.id === id) {
+              Object.assign(item, data);
```

Comment:
I think doing a `break` here would improve performance
```suggestion
              Object.assign(item, data);
              break;
```